doc/Overview/PropertySelector/concept.md

```diff
@@ -0,0 +1,86 @@
+---
+uid: Overview.PropertySelector.Concept
+---
+# Concept
+
+The `PropertySelector` is a standardized way to declare a _path_ to a _value_ given from a root _entity_.
+Using this _path_, an helper _IValueAccessor_ is being generated at compile time to give read and write access to the target _value_.
+
+This is an helper to avoid the usage a 2 delegates (one for read, and one for write) to edit a _value_ on a given _entity_.
+
+For instance, given a `Movie` record:
+```csharp
+public partial record Movie(int Likes);
+```
+
+And an helper class:
+```csharp
+public static class Math
+{
+	public T Increment<T>(T instance, PropertySelector<T, int> selector)
+	{
+		// ..
+	}
+}
+```
+
+You can do something like:
+```
+var current = new Movie(0);
+var updated = Math.Increment(current, m => m.Likes);
+```
+
+In this example `current.Likes` will be `0` while `updated.Likes` is `1`.
+
+## Limitations
+
+1. Only records are supported.
+2. You can use only property in the delegate. Methods, constant or any other constructs are not supported.
+3. You cannot use method-group, only the lambda syntax is allowed.
+4. You delegate cannot use any capture.
+
+Basically you can only write something like `e => e.A.B.C` as `PropertySelector`.
+
+## Declare a PropertySelector parameter on a method
+
+To avoid usage of reflection at runtime, the `PropertySelector` relies on generated code to work.
+Considering this we need to match an instance of a `PropertySelector` at runtime to its declaration using an identifier 
+derived from information that are also avaliable at compile time.
+
+To avoid to require from the end user to profide a such unique identifier, we are relying on the `[CallerFilePath]` and `[CallerLineNumber]` attributes.
+When method parameters are flagged with those attributes, values are automatically full-filled by the compiler.
```

Comment:
Add documentation links to [CallerFilePath] and [CallerLineNumber].

```suggestion
To avoid requiring from the end user to provide a unique identifier, the generation tooling relies on the `[CallerFilePath]` and `[CallerLineNumber]` attributes.
When method parameters are flagged with those attributes, values are automatically populated by the compiler.
```